parse_str_test.go

```diff
@@ -38,28 +56,27 @@ func ExampleParseStr() {
 func ExampleParseStr_complexArray() {
 	// Each empty key will be treated as auto-incremented integer key for each
 	// array
-	fmt.Println(ParseStr("key=value&a[]=123&a[]=false&b[]=str&c[]=3.5&a[]=last"))
+	fmt.Println(convertOrderedMapToMap(ParseStr("key=value&a[]=123&a[]=false&b[]=str&c[]=3.5&a[]=last")))
 
 	// You can mix multiple types of keys in one dictionary, and you can mix
 	// empty key with non-empty key. Each non-empty integer key will be used as
 	// a new starting number for next empty key.
-	fmt.Println(ParseStr("arr[]=A&arr[]=B&arr[9]=C&arr[]=D&arr[foo]=E&arr[]=F&arr[15.1]=G&arr[]=H"))
+	fmt.Println(convertOrderedMapToMap(ParseStr("arr[]=A&arr[]=B&arr[9]=C&arr[]=D&arr[foo]=E&arr[]=F&arr[15.1]=G&arr[]=H")))
```

Comment:
The result of `input` on this line is **different for go version >= 1.21.**
```go
	map_parsestr := gophplib.ParseStr("arr[]=A&arr[]=B&arr[9]=C&arr[]=D&arr[foo]=E&arr[]=F&arr[15.1]=G&arr[]=H")
	fmt.Println(map_parsestr)

	init_map := map[any]any{
        "arr": map[any]any{            
            0:      "A",
            1:      "B",
            9:      "C",
            10:     "D",
            "foo":  "E",
            11:     "F",
			"15.1": "G",
            12:     "H",
        },
    }
	fmt.Println(init_map)
```
The images below are examples of how the same input is output differently in GO 1.20 and 1.22, respectively.

<img width="1486" alt="스크린샷 2024-03-25 오후 5 17 28" src="https://github.com/portone-io/gophplib/assets/7577827/063d657b-9576-4203-973d-94c7b719bbe0">
It seems that the value is always constant when the `map` is initialized, so it is probably sorted by some condition, but I haven't found anything about it.

So my suggestion is to modify this logic so that it doesn't work, but leave a `NOTE:` in the record.